tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -650,6 +647,74 @@ def func(x, y):
         assert np.allclose(results1, results3)
 
 
+# pylint: disable=import-outside-toplevel, not-an-iterable
+@pytest.mark.jax
+@pytest.mark.parametrize("diff_method", [None, "best"])
+@pytest.mark.parametrize("postselect", [None, 1])
+@pytest.mark.parametrize("reset", [False, True])
+def test_jax_jit(diff_method, postselect, reset):
+    """Tests that DefaultQubit handles a circuit with a single mid-circuit measurement and a
+    conditional gate. A single measurement of a common observable is performed at the end."""
+    import jax
+
+    shots = 10
+
+    dev = qml.device("default.qubit", shots=shots, seed=jax.random.PRNGKey(678))
+    params = [np.pi / 2.5, np.pi / 3, -np.pi / 3.5]
+    obs = qml.PauliY(0)
+
+    @qml.qnode(dev, diff_method=diff_method)
+    def func(x, y, z):
+        m0, m1 = obs_tape(x, y, z, reset=reset, postselect=postselect)
+        return (
+            # qml.probs(wires=[1]), # JAX cannot compile code calling qml.math.unique
+            qml.sample(wires=[1]),
+            qml.sample(wires=[0, 1]),
+            qml.expval(obs),
+            # qml.probs(obs), # JAX cannot compile code calling qml.math.unique
+            qml.sample(obs),
+            qml.var(obs),
+            qml.expval(op=m0 + 2 * m1),
+            qml.probs(op=m0),
+            qml.sample(op=m0 + 2 * m1),
+            qml.var(op=m0 + 2 * m1),
+            qml.probs(op=[m0, m1]),
+        )
+
+    func1 = func
+    results1 = func1(*params)
+
+    jaxpr = str(jax.make_jaxpr(func)(*params))
+    if diff_method == "best":
+        assert "pure_callback" in jaxpr
+        pytest.xfail("QNode with diff_method='best' cannot be compiled with jax.jit.")
+    else:
+        assert "pure_callback" not in jaxpr
+
+    func2 = jax.jit(func)
+    results2 = func2(*jax.numpy.array(params))
+
+    measures = [
+        # qml.probs,
```

Comment:
Do we want to remove it definitely? Or maybe comment why this is commented?

Reply:
Same below.

Reply:
Will be fixed in a follow-up PR.